packages/react-instantsearch/src/connectors/connectGeoSearch.js

```diff
@@ -0,0 +1,138 @@
+import createConnector from '../core/createConnector';
+import {
+  getResults,
+  getCurrentRefinementValue,
+  getIndex,
+  refineValue,
+  cleanUpValue,
+} from '../core/indexUtils';
+
+/**
+ * The GeoSearch connector provides the logic to build a widget that will display the results on a map.
+ * It also provides a way to search for results based on their position. The connector provides function to manage the search experience (search on map interaction).
+ * @name connectGeoSearch
+ * @kind connector
+ * @requirements Note that the GeoSearch connector uses the [geosearch](https://www.algolia.com/doc/guides/searching/geo-search) capabilities of Algolia.
+ * Your hits **must** have a `_geoloc` attribute in order to be passed to the rendering function.
+ * Currently, the feature is not compatible with multiple values in the `_geoloc` attribute.
```

Comment:
Maybe add: "Example: a restaurant in multiple locations". If you are in this case, you can deduplicate records and use distinct (link to doc). wdyt?